messaging/filter/service.go

```diff
@@ -137,9 +148,29 @@ func (s *Service) Init(chats []*Chat) ([]*Chat, error) {
 	return allChats, nil
 }
 
+func (s *Service) Start(checkPeriod time.Duration) {
+	ticker := time.NewTicker(checkPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			messages := s.getMessages()
+
+			if len(messages) != 0 {
+				s.onNewMessages(messages)
+			}
+		case <-s.quit:
+			return
+		}
+	}
+}
+
 // Stop removes all the filters
 func (s *Service) Stop() error {
 	var chats []*Chat
+
+	close(s.quit)
```

Comment:
Maybe we can put it at the top as it gets in the middle of collecting `chats`? 

Also, in theory, when `s.quite` is being closed, `<-ticker.C` could return and messages are processed. Not sure whether that's ok or we should wait for that loop to finish before we continue with removing chats.

Reply:
It should be ok, because those messages won't be confirmed and will just be fetched back once logging in again, if we add a lock as pointed above it should also take care of any race condition

Reply:
I mean,  I will put it at the top btw